src/main/java/com/desiato/puresynth/controllers/AudioController.java

```diff
@@ -19,24 +16,23 @@ public final class AudioController {
     private final AudioService audioService;
 
     @PostMapping("/generate")
-    public ResponseEntity<?> generateAudio(@RequestBody final AudioRequest request) {
+    public ResponseEntity<String> generateAudio(@RequestBody final AudioRequest request) {
+
         try {
+            // Generate the audio file as a byte array
             byte[] audioBytes = audioService.generateSineWaveFile(
                     request.getFrequency(),
                     request.getDuration());
 
-            ByteArrayResource resource = new ByteArrayResource(audioBytes);
+            String fileName = "sine_wave_" + request.getFrequency() + "Hz";
+
+            String filePath = audioService.saveAudioFile(audioBytes, fileName);
```

Comment:
File name and file apth are implementation details, and shouldn't be handled by the controller. They should be encapsualted in some object like `AudiFileInfo`.